### PR TITLE
cl: compileSelectorExpr use better lookup method

### PR DESCRIPTION
**Pull request by visualfc**
use findMethod toptr to build GoPackage symbol ` (struct).method or (*struct).method`
```
// t is struct or interface
func findMethod(t reflect.Type, name string) (method reflect.Method, toptr bool, found bool) {
	method, found = t.MethodByName(name)
	if !found && t.Kind() == reflect.Struct {
		t = reflect.PtrTo(t)
		toptr = true
		method, found = t.MethodByName(name)
	}
	return
}
```

### CODE REVIEW

File: cl/expr_test.go [hunk 1]
```diff
@@ -833,6 +833,20 @@ func TestPkgMethod(t *testing.T) {
 	`, "int\n")
 }
 
+func TestPkgMethodBadCall(t *testing.T) {
+	cltest.Expect(t, `
+	import "bytes"
+	buf := bytes.NewBuffer([]byte("hello"))
+	println((&buf).String())
```
Comment (xushiwei) [inline cl/expr_test.go:840]: For the Go compiler, this example will be reported the following error:

```
calling method String with receiver &buf (type **bytes.Buffer) requires explicit dereference.
```

File: cl/expr_test.go [hunk 2]
```diff
@@ -833,6 +833,20 @@ func TestPkgMethod(t *testing.T) {
...
+	`, "", nil)
+	cltest.Expect(t, `
+	import "reflect"
+	v := reflect.ValueOf(100)
+	p := &v
+	println((&p).Kind())
```
Comment (xushiwei) [inline cl/expr_test.go:846]: For the Go compiler, this example will be reported the following error:

```
calling method Kind with receiver &p (type **reflect.Value) requires explicit dereference.
```

Comment (visualfc) [reply cl/expr_test.go:846]: fixed, clean ctx.code  ast.ParenExpr  (&p) -> &p

**Comment by xushiwei** [comment]
please fix conflictions.

**Pull request merged**